python/lsst/pipe/base/graph/_implDetails.py

```diff
@@ -181,3 +233,119 @@ def __contains__(self, key: _T) -> bool:
             Boolean of the presence of the supplied key
         """
         return key in self._producers or key in self._consumers
+
+
+def _pruner(datasetRefDict: _DatasetTracker[DatasetRef, QuantumNode], refsToRemove: Iterable[DatasetRef], *,
+            alreadyPruned: Optional[Set[QuantumNode]] = None):
+    r"""Prunes supplied dataset refs out of datasetRefDict container, recursing
```

Comment:
```suggestion
    r"""Prune supplied dataset refs out of datasetRefDict container, recursing
```
(docstrings should be in imperative voice)